packages/expo-image/ios/EXImageView.m

```diff
@@ -17,9 +36,67 @@ - (void)dealloc
 
 - (void)setSource:(NSDictionary *)source
 {
-  NSURL *imageURL = [RCTConvert NSURL:source[sourceUriKey]];
+  _source = source;
+  _needsReload = YES;
+}
 
-  [self sd_setImageWithURL:imageURL];
+- (void)setResizeMode:(RCTResizeMode)resizeMode
+{
+  if (_resizeMode == resizeMode) return;
+  
+  // Image needs to be reloaded whenever repeat is enabled or disabled
+  _needsReload = _needsReload || (resizeMode == RCTResizeModeRepeat) || (_resizeMode == RCTResizeModeRepeat);
+  _resizeMode = resizeMode;
+  
+  // Repeat resize mode is handled by the UIImage. Use scale to fill
+  // so the repeated image fills the UIImageView.
+  self.contentMode = resizeMode == RCTResizeModeRepeat
```

Comment:
Maybe we should move this change to the completion block of the `sd_setImageWithURL`?

Reply:
No that wouldn't work. The image post-process function only needs to be called whenever the repeat-mode changes (or when the cap-insets change, but that will be added in a later PR)

Reply:
True, let me rephrase my concern:

If we set the `contentMode` immediately when the prop is set, on a slow device in the following scenario we would see a glitch:

- an image `X` is downloaded and presented with `resizeMode = A`
- props change: image changes to `Y`, `resizeMode` changes to `B`
  - `setSource:` is called, source updates, `needsReload` is set to `true`
  - `setResizeMode:` is called, `needsReload` is set to `true`, **`contentMode` is changed immediately**
  - `didSetProps:` is called, a new image is downloaded and presented

For a brief moment image `X` is presented with `contentMode` `B`.

Reply:
I understand your concern here and your point is entirely valid.

However, the issue of changing props and then consistently updating the view in one pass (synchronously) would still remain for other props like borders, shadows, etc.. It would just be solved for this particular case..

If you would want to solve this right, you would have to sync all the prop updates on the view (e.g. after the image load has completed). However, this would also affect animatable props and probably make them much update much slower. Personally, I don't think this is a route we would want to go down into.

Reply:
Sorry for not making myself clear, what I wanted to propose was to _solve things right_ (as you named it) and sync all the prop updates. 😅 When it comes to animatable properties I don't see why would the implementation be slower for them (the only difference that I see would be that in `didSetProps` we would have to compare more values: previous vs next border radii, shadow, etc, which shouldn't be that much slower 🤔).

Looking into other image components (`RCTImageView` and `FFImageView`) I can see they both implement `setResizeMode:` the same way (set the value immediately), so maybe we could get away with it… but I certainly wish we could make sure the scenario above wouldn't happen.

Would you care to add a comment like `// TODO (@sjchmiela): Research props syncing <link to discussion or issue #7440>` so it reminds me to get back to this issue when we're done with major features?